src/apigateway/commands/invokeRemoteRestApi.ts

```diff
@@ -0,0 +1,188 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import { RestApiNode } from '../explorer/apiNodes'
+import { getLogger, Logger } from '../../shared/logger'
+import { BaseTemplates } from '../../shared/templates/baseTemplates'
+import { ext } from '../../shared/extensionGlobals'
+import _ = require('lodash')
+import { toArrayAsync, toMap } from '../../shared/utilities/collectionUtils'
+import { ExtensionUtilities } from '../../shared/extensionUtilities'
+import { ApigTemplates } from '../templates/apigTemplates'
+import { Resource } from 'aws-sdk/clients/apigateway'
+import { ApiGatewayClient } from '../../shared/clients/apiGatewayClient'
+
+export async function invokeRemoteRestApi(params: { outputChannel: vscode.OutputChannel; apiNode: RestApiNode }) {
+    const logger: Logger = getLogger()
+    const apiNode = params.apiNode
+
+    try {
+        const view = vscode.window.createWebviewPanel(
+            'html',
+            `Invoke methods for ${apiNode.name}`,
+            vscode.ViewColumn.One,
+            {
+                // Enable scripts in the webview
+                enableScripts: true,
+            }
+        )
+        const baseTemplateFn = _.template(BaseTemplates.SIMPLE_HTML)
+
+        view.webview.html = baseTemplateFn({
+            cspSource: view.webview.cspSource,
+            content: '<h1>Loading...</h1>',
+        })
+
+        const invokeTemplateFn = _.template(ApigTemplates.REMOTE_INVOKE_TEMPLATE)
+
+        const client = ext.toolkitClientBuilder.createApiGatewayClient(params.apiNode.regionCode)
+        logger.info(`Loading API Resources for API ${apiNode.name} (id: ${apiNode.id})`)
+
+        try {
```

Comment:
This nested try within a try is interesting. Can it be moved to a separate method?